File: catering_module/public/js/bom_addon.js
frappe.ui.form.on('BOM Item', {
	qty_recipe: function(frm,cdt,cdn) {
		var child = locals[cdt][cdn];
        if(child.item_code){
            frappe.call({
                method: 'frappe.client.get_value',
                args: {
                  doctype: 'Item',
                  filters: { name: child.item_code },
                  fieldname: '*'
                },
                callback: function(response) {
                    var data = response.message;
                    var q_factor = data.q_factor;
                    var q_req = child.qty_recipe * (1 + (q_factor/100));
                    frappe.model.set_value(cdt, cdn, "qty", q_req);
                }
              });
        }
	},
});

File: catering_module/public/bom_addon.py
import frappe
from frappe.model.document import Document
import json
import math

def validate_stock_qty_rounded(doc,name):
    for i in doc.items:
        rounding_up = frappe.get_value("Item",i.item_code,"kelipatan_rounding_up")
        x = i.qty / rounding_up
        x = math.ceil(x)
        hasil = x * rounding_up
        i.stock_qty = hasil